server/data/reviews.js

```diff
@@ -0,0 +1,274 @@
+const mongoCollections = require('../config/mongoCollections')
+const errors = require('./errors')
+const showCollection = mongoCollections.shows
+const movieCollection = mongoCollections.movies
+const userCollection = mongoCollections.users
+const { ObjectId } = require('mongodb')
+const { reviews } = require('../config/mongoCollections')
+
+// Create a new review
+const createReview = async (
+  reviewerId,
+  reviewer,
+  contentId,
+  dateOfReview,
+  review,
+  like_dislike
+) => {
+  let inserted
+  reviewerId = errors.validateObjectId(reviewerId)
+  reviewer = errors.validateReviewer(reviewer)
+  contentId = errors.validateObjectId(contentId)
+  dateOfReview = errors.validateDateOfReview(dateOfReview)
+  review = errors.validateReview(review)
+  _id = new ObjectId()
+
+  shows = await showCollection()
+  show = await shows.findOne({ _id: contentId })
+
+  const movies = await movieCollection()
+  const movie = await movies.findOne({ _id: contentId })
+
+  const users = await userCollection()
+  const user = await users.findOne({ _id: reviewerId })
+
+  const newReview = {
+    _id: _id,
+    reviewerId: reviewerId,
+    reviewer: reviewer,
+    dateOfReview: dateOfReview,
+    contentId: contentId,
+    review: review,
+    like_dislike: like_dislike,
+  }
+
+  if (!movie && !show) {
+    throw 'content does not exist'
+  }
+
+  if (!movie) {
+    inserted = await shows.updateOne(
+      { _id: contentId },
+      { $push: { reviews: newReview } }
+    )
+    if (inserted.modifiedCount === 0) throw 'Could not update the show'
+  }
+
+  if (!show) {
+    inserted = await movies.updateOne(
+      { _id: contentId },
+      { $push: { reviews: newReview } }
+    )
+    if (inserted.modifiedCount === 0) throw 'Could not update the movie'
+  }
+
+  inserted = await users.updateOne(
+    { _id: reviewerId },
+    { $push: { reviews: newReview } }
+  )
+  if (inserted.modifiedCount === 0) throw 'Could not update the shows'
+  const createdReview = await users.findOne({ 'reviews.contentId': contentId })
+
+  if (!movie) {
+    shows = await showCollection()
+    show = await shows.findOne({ _id: contentId })
+    const percent = overallRating(show, contentId)
+    const updateRating = await shows.updateOne(
+      { _id: contentId },
+      { $set: { overallRating: Number(percent) } }
+    )
+  }
+
+  if (!show) {
+    const movies = await movieCollection()
+    const movie = await movies.findOne({ _id: contentId })
+    const percent = overallRating(movie, contentId)
+    const updateRating = await movies.updateOne(
+      { _id: contentId },
+      { $set: { overallRating: Number(percent) } }
+    )
+  }
```

Comment:
Well in this case I cannot do it in the same update. I tried doing that but while calculating the ratings it will still calculate based on the old data unless a review is inserted with the likes and dislikes and then called again from the database